include/range/v3/at.hpp

```diff
@@ -14,35 +15,64 @@
 #ifndef RANGES_V3_AT_HPP
 #define RANGES_V3_AT_HPP
 
+#include <stdexcept>
 #include <range/v3/range_fwd.hpp>
 #include <range/v3/begin_end.hpp>
 #include <range/v3/range_traits.hpp>
 #include <range/v3/range_concepts.hpp>
-#include <range/v3/utility/iterator.hpp>
+#include <range/v3/distance.hpp>
 #include <range/v3/utility/static_const.hpp>
+#include <range/v3/index.hpp>
+
+RANGES_DISABLE_WARNINGS
+RANGES_DIAGNOSTIC_IGNORE_SIGN_CONVERSION
 
 namespace ranges
 {
     inline namespace v3
     {
+        /// Checked indexed range access.
+        ///
         /// \ingroup group-core
         struct at_fn
         {
             /// \return `begin(rng)[n]`
             template<typename Rng,
-                CONCEPT_REQUIRES_(RandomAccessRange<Rng>())>
+                CONCEPT_REQUIRES_(RandomAccessRange<Rng>() && SizedRange<Rng>())>
             RANGES_CXX14_CONSTEXPR
-            auto operator()(Rng &&rng, range_difference_type_t<Rng> n) const ->
+            auto operator()(Rng &&rng, range_size_type_t<Rng> n) const ->
```

Comment:
> I've fixed it using your first solution.

I completely failed to express that the implicit conversion wrapper was for use in the return expression a la `begin()[wrapper<range_difference_type_t<Rng>>{n}]` to avoid conversion warnings if someone defines `operator[]` for their iterator as taking an unsigned type. That said, I can't imagine *why* someone would do so or why we should encourage such a thing. Consider that suggestion withdrawn.